resource/labeling-app/components/image-object-sample/AnnotationThumb.js

```diff
@@ -0,0 +1,42 @@
+let AnnotationThumb = {
+    props: {
+        data: Object,
+        color: Array
+    },
+    watch: {
```

Comment:
yes, I generally agree, but in this case, it's going to become a bit cumbersome since we can't directly access this.$el inside `computed` (i'll be undefined) and we'll have to initialize clientWidth and clientHeight in a `mounted` callback (thus define them in the `data`)
All in all, it's possible, but since it doesn't seem to be a big violation of Vue rules I'd keep it like it is